torch/nn/_intrinsic/qat/modules/conv_bn_relu.py

```diff
@@ -0,0 +1,184 @@
+        self.num_batches_tracked.zero_()
+
+    def reset_parameters(self):
+        self.reset_running_stats()
+        if self.affine:
+            init.uniform_(self.gamma)
+            init.zeros_(self.beta)
+
+    def _forward(self, input):
+        # exponential_average_factor is self.momentum set to
+        # (when it is available) only so that if gets updated
+        # in ONNX graph when this node is exported to ONNX.
+        if self.momentum is None:
+            exponential_average_factor = 0.0
+        else:
+            exponential_average_factor = self.momentum
+
+        if self.training and self.track_running_stats:
+            # TODO: if statement only here to tell the jit to skip emitting this when it is None
+            if self.num_batches_tracked is not None:
+                self.num_batches_tracked += 1
+                if self.momentum is None:  # use cumulative moving average
+                    exponential_average_factor = 1.0 / float(self.num_batches_tracked)
+                else:  # use exponential moving average
+                    exponential_average_factor = self.momentum
+        scaled_weight = self.gamma * self.weight / torch.sqrt(self.running_var)
+        conv = conv2d_forward(input, self.padding_mode, self.padding,
+                              self.weight_fake_quant(scaled_weight), self.bias,
+                              self.stride, self.dilation, self.groups)
+        batch_mean = torch.mean(conv, dim=[0, 2, 3])
+        n = input.numel() / self.num_features
+        batch_var = torch.var(conv, dim=[0, 2, 3]) * (n / (n - 1))
+        self.running_mean = exponential_average_factor * batch_mean + (1 - exponential_average_factor) * self.running_mean
+        self.running_var = exponential_average_factor * batch_var + (1 - exponential_average_factor) * self.running_var
+
+        if not self.freeze_bn:
+            conv *= torch.sqrt(self.running_var) / torch.sqrt(batch_var)
+            conv += self.beta - self.gamma * (batch_mean / torch.sqrt(batch_var))
+        else:
+            conv += self.beta - self.gamma * self.running_mean / torch.sqrt(self.running_var)
+
+        return conv
+
+    def forward(self, input):
+        return self.observer(self._forward(input))
+
+    @classmethod
+    def from_float(cls, mod, qconfig):
+        r"""Create a qat module from a float module or qparams_dict
+
+            Args: `mod` a float module, either produced by torch.quantization utilities
+            or directly from user
+        """
+        assert type(mod) == cls.__FLOAT_MODULE__, ' qat.' + cls.__name__ + '.from_float only works for ' + \
+            cls.__FLOAT_MODULE__.__name__
+        if not qconfig:
+            assert hasattr(mod, 'qconfig'), 'Input float module must have qconfig defined'
+            assert hasattr(mod, 'observer'), 'Input float module must have observer attached'
+            qconfig = mod.qconfig
+        qat_convbn = cls(mod.in_channels, mod.out_channels, mod.kernel_size,
```

Comment:
Is the input to from_float a single sequential module? In that case, this code should refer to mod[0] and mod[1]. 
Also, please add a simple test that shows that numerically the forward and backward of this module matches that of a conv followed by a bn.

Reply:
I think we can access all attributes of submodules in sequential?

yeah plan to add numerical tests

Reply:
What I was saying was that the from_float function should do the following (note the indices into the sequential container)
```PYTHON
    def from_float(cls, mod, qconfig):
        r"""Create a qat module from a float module or qparams_dict
            Args: `mod` a float module, either produced by torch.quantization utilities
            or directly from user
        """
        assert type(mod) == cls.__FLOAT_MODULE__, ' qat.' + cls.__name__ + '.from_float only works for ' + \
            cls.__FLOAT_MODULE__.__name__
        if not qconfig:
            assert hasattr(mod, 'qconfig'), 'Input float module must have qconfig defined'
            assert hasattr(mod, 'observer'), 'Input float module must have observer attached'
            qconfig = mod.qconfig
        qat_convbn = cls(mod[0].in_channels, mod[0].out_channels, mod[0].kernel_size,
                         stride=mod[0].stride, padding=mod[0].padding, dilation=mod[0].dilation,
                         groups=mod[0].groups, bias=mod[0].bias is not None,
                         padding_mode=mod[0].padding_mode,
                         eps=mod[1].eps, momentum=mod[1].momentum,
                         freeze_bn=False,
                         activation_fake_quant=qconfig.activation,
                         weight_fake_quant=qconfig.weight)

        qat_convbn.weight = mod[0].weight
        qat_convbn.bias = mod[0].bias
        qat_convbn.gamma = mod[1].gamma
        qat_convbn.beta = mod[1].beta
        qat_convbn.running_mean = mod[1].running_mean
        qat_convbn.running_var = mod[1].running_var
        qat_convbn.num_batches_tracked = mod[1].num_batches_tracked
        return qat_convbn
```